src/main/java/com/blackducksoftware/integration/alert/channel/hipchat/HipChatChannel.java

```diff
@@ -126,25 +131,79 @@ public String testGlobalConfig(final HipChatGlobalConfigEntity entity) throws In
     }
 
     @Override
-    public Request createRequest(final ChannelRequestHelper channelRequestHelper, final HipChatDistributionConfigEntity config, final HipChatGlobalConfigEntity globalConfig, final DigestModel digestModel) throws IntegrationException {
+    public List<Request> createRequests(final ChannelRequestHelper channelRequestHelper, final HipChatDistributionConfigEntity config, final HipChatGlobalConfigEntity globalConfig, final ChannelEvent event)
+            throws IntegrationException {
         if (config.getRoomId() == null) {
             throw new IntegrationException("Room ID missing");
         } else {
-            final Collection<ProjectData> projectDataCollection = digestModel.getProjectDataCollection();
-            final String htmlMessage = createHtmlMessage(projectDataCollection);
-            final String jsonString = getJsonString(htmlMessage, AlertConstants.ALERT_APPLICATION_NAME, config.getNotify(), config.getColor());
+            if (isChunkedMessageNeeded(event)) {
+                final String eventContent = event.getContent();
+                final int requestCount = calculateChunks(eventContent);
+                final List<Request> requestList = createChunkedRequestList(channelRequestHelper, config, globalConfig, event, requestCount);
+                return requestList;
+            } else {
+                final String contentTitle = String.format("%s -> %s", event.getProvider(), event.getNotificationType());
+                return Arrays.asList(createRequest(channelRequestHelper, config, globalConfig, contentTitle, event.getContent()));
+            }
+        }
+    }
 
-            final String url = getApiUrl(globalConfig) + "/v2/room/" + config.getRoomId().toString() + "/notification";
+    private boolean isChunkedMessageNeeded(final ChannelEvent event) {
+        final String eventContent = event.getContent();
+        if (StringUtils.isNotBlank(eventContent)) {
+            return eventContent.length() > MESSAGE_SIZE_LIMIT;
+        } else {
+            return false;
+        }
+    }
 
-            final Map<String, String> requestHeaders = new HashMap<>();
-            requestHeaders.put("Authorization", "Bearer " + globalConfig.getApiKey());
-            requestHeaders.put("Content-Type", "application/json");
+    private int calculateChunks(final String eventContent) {
```

Comment:
Do we need this method for anything other than logging?

Reply:
The return value of the method is used to be able to split the content into pieces.  We also use the return value to include in the message to denote the chunk number in the message we send to hipchat (1 of 10) for example.